_posts/en/newsletters/2025-01-10-newsletter.md

```diff
@@ -0,0 +1,175 @@
+  prevent forwarding nodes from earning income on their capital (through
+  [channel jamming attacks][topic channel jamming attacks] and other
+  problems).
+
+  Sanders suggests solving the problem with a relative timelock that
+  would apply to all transactions required to settle a contract.  In
+  LN-Symmetry had such a feature and Alice got the penultimate state
+  confirmed, Bob would need to get the last state confirmed before the
+  deadline of the penultimate state.  In a [later post][sanders tpp],
+  Sanders links to a channel protocol by John Law (see [Newsletter
+  #244][news244 tpp]) that uses two transaction-level relative timelocks
+  to provide a contract-level relative timelock without consensus
+  changes.  However, that doesn't work for LN-Symmetry which allows each
+  state to spend from any previous state.
+
+  Sanders sketches a solution, but notes that it has problems.  He also
+  notes how the problem could be solved using Chia's `coinid` feature,
+  which appears to be similar to John Law's 2021 idea for Inherited
+  Identifiers (IIDs).  Jeremy Rubin [replied][rubin muon] with a link to
+  his proposal last year for _muon_ outputs that must be spent in the
+  same block as the transaction that created them, showing how they
+  could contribute to a solution.  Sanders mentions, and Anthony Towns
+  [expands][towns coinid] on, the `coinid` feature from the Chia
+  blockchain, showing how it could reduce the amount of data required
+  down to a constant amount.  Salvatore Ingala [posted][ingala cat]
+  about a similar mechanism using [OP_CAT][topic op_cat] that he learned
+  about from developer Rijndael, who later [provided details][rijndael
+  cat] himself.  Brandon Black [described][black penalty] an
+  alternative type of solution---a penalty-based variant of
+  LN-Symmetry---and cited work by Daniel Roberts about it (see next news
+  item).
+
+- **Multiparty LN-Symmetry variant with penalties for limiting published updates:**
+  Daniel Roberts [posted][roberts sympen] to Delving Bitcoin about
+  preventing a malicious channel counterparty (Mallory) from being able
+  to delay channel settlement by deliberately broadcasting old states at
+  a higher feerate than an honest counterparty (Bob) is paying for
+  confirmation of the final state.  In theory, Bob can rebind his final
+  state to Mallory's old state and both transactions might confirm in
+  the same block, causing her to lose money on fees and Bob to get the
+  final state confirmed for the same fee cost he was already willing to
+  pay.  However, if Mallory can repeatedly prevent Bob from learning
+  about her broadcasts of old states before they get confirmed, she can
+  prevent him from responding until an [HTLCs][topic htlc] in the
+  channel expire and Mallory is able to steal funds.
+
+  Roberts proposed a scheme that only allows a channel participant to
+  get a single state confirmed.  If a later state gets confirmed, the
```

Comment:
nit, the frequent use of "gets confirmed" seems clunky, maybe prefer one of "is confirmed" / "have a state confirmed" / "<party> confirms" etc.

Reply:
Agree it's clunky and I'll try to fix these.  I don't think it's important here (which is why I'm changing it), but I habitually use that language because confirming something isn't any individual's choice (unless they have 100% hashrate); it's probalistically determined based on overall miner choices.  C.f., I can't add code to Bitcoin Core (because I don't have commit access) but I can _get_ code added to Bitcoin Core (by following the regular process that ends with maintainer merging).